tokio/src/net/tcp/stream.rs

```diff
@@ -826,6 +886,12 @@ mod sys {
             self.io.as_raw_fd()
         }
     }
+
+    impl From<&TcpStream> for mio::net::TcpSocket {
```

Comment:
We cannot impl conversion traits from `mio` types as that would expose `mio` as part of the public API.

I would suggest just adding private `to_mio(&self) -> mio::net::TcpSocket`.

Reply:
added